src/enrichment.h

```diff
@@ -324,10 +324,10 @@ class Enrichment
   double initial_feed;
 
   #pragma cyclus var {							\
-    "default": 1e299, "tooltip": "max inventory of feed material (kg)", \
+    "default": "cyclus::cy_large_double", "tooltip": "max inventory of feed material (kg)", \
     "uilabel": "Maximum Feed Inventory", \
     "uitype": "range", \
-    "range": [0.0, 1e299], \
+    "range": [0.0, "cyclus::cy_large_double"], \
```

Comment:
So this has me thinking about where else this metadata is used...  If you have an installed build with this branch, can you run `cyclus -m` and see what it reports as the range?  Has the substitution happened at that point? 

Other tools may read/parse this metadata without access to `cyclus::cy_large_double`

Reply:
The substitution has not happened at that point 
![image](https://github.com/cyclus/cycamore/assets/79653949/14bac0a3-a60b-47a6-bb2d-63324ec5718b)
Do we want those values to be expanded to their numeric equivalents? I'll see if it can be done given the build process in place

Reply:
There are two issues here... 
1. on one hand it would be good to have expanded so that other users of this metadata know what it is
2. if we want modules to be plug-an-play with different installs of Cyclus, maybe we want this to be substituted later??

It's possible we could do both, but performing the substitution before we publish the metadata?  That might be a bigger project though!

Reply:
What if we substitute later and have a section of metadata (or a different cli flag perhaps) that shows users the values for constants in this install of cyclus?  That way we don't need to modify the preprocessor, modules are plug-and-play, and users have a way to see what the constants are (as long as we put them in an intuitive place)

Reply:
Related - how is the substitution made under your current paradigm - when is it substituted and what can it be used for?

Reply:
Right now `cyclus::cy_large_double` is a `static const double` so it is evaluated at runtime and exists in the cyclus C++ namespace (and in the `cyclus.lib` python library via cython wrapper).  It is not evaluated before/by `cycpp.py`.  We could probably `#define` the value instead so it is substituted by `cpp` (though I'm still not sure there's a way to substitute it into the `#pragma cyclus` statement to be evaluated before `cycpp.py` is run).  Does that answer your question?

Reply:
I'm trying to recall if there is any code on the Cyclus side that uses those values and were the substitution happens to use those values.  For example, if the `range` of a variable is set in these pragmas, I think it results in an XML grammar entry that requires the value to be within that range.  Is the substitution done before generating the snippet of grammar so that XML only knows about the numerical value?  Or is the substitution happening later?

It's also possible that the value is not actually used anywhere by Cyclus and only exists for other tools that use the metadata.

Reply:
`KFacility`, `Sink`, and `Source` in cyclus use this value.  When calling `cyclus -m` the variable name is still included in metadata instead of the value.  However, when viewing an agent's xml schema I don't see any reference at all.  For example - `capacity` in `Source` is defined as so:
```
  #pragma cyclus var { \
    "doc": "amount of commodity that can be supplied at each time step", \
    "uilabel": "Maximum Throughput", \
    "uitype": "range", \
    "range": [0.0, "cy_large_double"], \
    "tooltip": "source capacity" \
  }
  double capacity;
```

But this is what I see when printing the agent schema via the cyclus cli:
```
root@a52714cb3b92:/cyclus# cyclus --agent-schema :agents:Source
<interleave>
    <element name="commod">
        <data type="token"/>
    </element>
    <optional>
        <element name="recipe_name">
            <data type="token"/>
        </element>
    </optional>
    <element name="capacity">
        <data type="double"/>
    </element>
</interleave>
```

The section of JSON dedicated to `Source` when calling `cyclus -m` is:
```
 ":agents:Source": {
   "all_parents": [
    "cyclus::Agent", 
    "cyclus::Facility", 
    "cyclus::Ider", 
    "cyclus::StateWrangler", 
    "cyclus::TimeListener", 
    "cyclus::Trader"
   ], 
   "doc": "A minimum implementation source facility that provides a commodity with a given capacity", 
   "entity": "facility", 
   "name": "cyclus::Source", 
   "parents": ["cyclus::Facility"], 
   "vars": {
    "capacity": {
     "alias": "capacity", 
     "doc": "amount of commodity that can be supplied at each time step", 
     "index": 2, 
     "range": [0.0, "cy_large_double"], 
     "shape": [-1], 
     "tooltip": "source capacity", 
     "type": "double", 
     "uilabel": "Maximum Throughput", 
     "uitype": "range"
    }, 
    "commod": {
     "alias": "commod", 
     "doc": "commodity that the source facility supplies", 
     "index": 0, 
     "schematype": "token", 
     "shape": [-1], 
     "tooltip": "source commodity", 
     "type": "std::string", 
     "uilabel": "Commodity", 
     "uitype": "outcommodity"
    }, 
    "recipe_name": {
     "alias": "recipe_name", 
     "default": "", 
     "doc": "Recipe name for source facility's commodity.If empty, source supplies material with requested compositions.", 
     "index": 1, 
     "schematype": "token", 
     "shape": [-1], 
     "tooltip": "commodity recipe name", 
     "type": "std::string", 
     "uilabel": "Recipe", 
     "uitype": "recipe"
    }
   }
```